lib/jackalope/session.ex

```diff
@@ -217,20 +222,22 @@ defmodule Jackalope.Session do
     # Setup the options for the work order; so far we support time to
     # live, which allow us to specify the time a work order is allowed
     # to stay in the work list before it is deemed irrelevant
-    expiration =
-      case Keyword.get(opts, :ttl, :infinity) do
-        :infinity -> :infinity
-        ttl when is_integer(ttl) -> System.monotonic_time(:millisecond) + ttl
-      end
-
-    # Note that we don't really concern ourselves with the order of
-    # the commands; the work_list is a list (and thus a stack) and when
-    # we retry a message it will reenter the work list at the front,
-    # and it could already have messages, etc.
-    work_list_opts = [expiration: expiration]
-    work_list = [{cmd, work_list_opts} | work_list]
-    state = %State{state | work_list: bound(work_list, state.max_work_list_size)}
-    {:noreply, state, {:continue, :consume_work_list}}
+    ttl = Keyword.get(opts, :ttl, @default_ttl_msecs)
+
+    if is_integer(ttl) and ttl >= 0 do
```

Comment:
```suggestion
    if is_integer(ttl) and ttl > 0 do
```
It seems less confusing to not support 0 TTL so that we don't have to define what that means.